libs/nucs/src/policy.rs

```diff
@@ -0,0 +1,316 @@
+            ConnectorPolicy::Or(conditions) => conditions.iter().any(|condition| condition.evaluate(value)),
+            ConnectorPolicy::Not(policy) => !policy.evaluate(value),
+        }
+    }
+}
+
+impl From<ConnectorPolicy> for Policy {
+    fn from(operator: ConnectorPolicy) -> Self {
+        Self::Connector(operator)
+    }
+}
+
+impl<'de> Deserialize<'de> for Policy {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_any(PolicyVisitor)
+    }
+}
+
+struct PolicyVisitor;
+
+impl PolicyVisitor {
+    fn build_selector<'de, A>(&self, seq: &mut A, encountered: usize) -> Result<Selector, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let selector = seq
+            .next_element::<String>()?
+            .ok_or_else(|| Error::invalid_length(encountered, &"an attribute selector"))?;
+        selector.parse().map_err(|e: SelectorParseError| Error::invalid_value(Unexpected::Seq, &e.to_string().as_str()))
+    }
+
+    fn build_simple_operator<'de, A>(self, mut seq: A, op: String) -> Result<OperatorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let selector = self.build_selector(&mut seq, 1)?;
+        let value = seq.next_element::<serde_json::Value>()?.ok_or_else(|| Error::invalid_length(2, &"a value"))?;
+        let operator = match op.as_str() {
+            "==" => Operator::Equals(value),
+            "!=" => Operator::NotEquals(value),
+            _ => panic!("unsupported operator"),
+        };
+        Ok(OperatorPolicy { selector, operator })
+    }
+
+    fn build_list_operator<'de, A>(self, mut seq: A, op: String) -> Result<OperatorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let selector = self.build_selector(&mut seq, 1)?;
+        let values = seq
+            .next_element::<Vec<serde_json::Value>>()?
+            .ok_or_else(|| Error::invalid_length(2, &"a list of values"))?;
+        let operator = match op.as_str() {
+            "anyOf" => Operator::AnyOf(values),
+            _ => panic!("unsupported operator"),
+        };
+        Ok(OperatorPolicy { selector, operator })
+    }
+
+    fn build_simple_connector<'de, A>(self, mut seq: A, connector: String) -> Result<ConnectorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let policy = seq.next_element::<Box<Policy>>()?.ok_or_else(|| Error::invalid_length(1, &"a policy"))?;
+        let connector = match connector.as_str() {
+            "not" => ConnectorPolicy::Not(policy),
+            _ => panic!("unsupported operator"),
+        };
+        Ok(connector)
+    }
+
+    fn build_list_connector<'de, A>(self, mut seq: A, connector: String) -> Result<ConnectorPolicy, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        let policies =
+            seq.next_element::<Vec<Policy>>()?.ok_or_else(|| Error::invalid_length(1, &"a list of policies"))?;
```

Comment:
Just to confirm, you truly want to return an error in cases like `[ "and", [] ]` ?

Reply:
This is up for discussion, I'm on the fence on this one. Maybe allowing this will give you more flexibility when building more complex conditions.